src/modules/github/components/pulls-review-dialog/pulls-review-dialog.tsx

```diff
@@ -0,0 +1,262 @@
+import {
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  ExternalLinkIcon
+} from 'lucide-react'
+import Link from 'next/link'
+import { type ChangeEvent, useEffect, useState } from 'react'
+
+import { Alert } from '@/elements/root/alert/alert'
+import { Button } from '@/elements/root/button/button'
+import { Dialog } from '@/elements/root/dialog/dialog'
+import { Progress } from '@/elements/root/progress/progress'
+import { RadioGroup } from '@/elements/root/radio-group/radio-group'
+import { Textarea } from '@/elements/root/textarea/textarea'
+import { Tooltip } from '@/elements/root/tooltip/tooltip'
+
+import { useDictionary } from '@/contexts/root/dictionary-provider/dictionary-provider'
+
+import { PullReviewDialogBody } from '@/components/github/root/pull-review-dialog-body/pull-review-dialog-body'
+
+import { useRepos } from '@/contexts/github/root/selected-pulls-provider'
+
+import { EPullRequestType } from '@/types/github/root/index'
+
+import { useSubmitForm } from './hooks'
+import { getFlattenCheckedPulls } from './utils'
+
+interface IPullsReviewDialogProps {
+  isDialogOpen: boolean
+  handleSetIsOpenDialog: (open: boolean) => void
+  handleSetHasChecked: (value: boolean) => void
+}
+
+export const PullsReviewDialog = ({
+  isDialogOpen,
+  handleSetIsOpenDialog,
+  handleSetHasChecked
+}: IPullsReviewDialogProps) => {
+  const [commentInput, setCommentInput] = useState<string>('')
+  const [focusIndex, setFocusIndex] = useState<number>(0)
+  const [radioButtonValue, setRadioButtonValue] = useState<EPullRequestType>(
+    EPullRequestType.COMMENT
+  )
+  const { repos } = useRepos()
+  const { translate } = useDictionary()
+  const {
+    progressData,
+    submit,
+    isLoading,
+    error: errors,
+    reset
+  } = useSubmitForm()
+  const radioButtonValues = [
+    {
+      value: EPullRequestType.COMMENT,
+      label: translate('GITHUB.PULL_REVIEW_FORM_COMMENT_BUTTON')
+    },
+    {
+      value: EPullRequestType.APPROVE,
+      label: translate('GITHUB.PULL_REVIEW_FORM_APPROVE_BUTTON')
+    },
+    {
+      value: EPullRequestType.REQUEST_CHANGES,
+      label: translate('GITHUB.PULL_REVIEW_FORM_REQUEST_CHANGES_BUTTON')
+    }
+  ]
+  const flattenCheckedPulls = getFlattenCheckedPulls(repos)
+  const focusedPull =
+    flattenCheckedPulls.length > 0 ? flattenCheckedPulls[focusIndex] : undefined
+
+  const handleRadioButtonChange = (value: string) => {
+    setRadioButtonValue(value as EPullRequestType)
+  }
+
+  const handleCommentChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
+    setCommentInput(event.target.value)
+  }
+
+  const handleOpenChange = (open: boolean) => {
+    setFocusIndex(0)
```

Comment:
No need to `setFocusIndex(0)` here. You can reset the state when the dialog is closed.

ref
```typescript
  useEffect(() => {
    if (!isDialogOpen) {
      reset()
    }
  }, [isDialogOpen])

```